app/models/story.rb

```diff
@@ -70,11 +76,10 @@ def set_last_update_time!(time=nil)
   end
 
   def self.for_user(user)
-    if user.nil? or user.sfw_filter
-      where("NOT adult").joins("LEFT OUTER JOIN watchlists ON watchlists.id = stories.watchlist_id").where("watchlists.id IS NULL OR watchlists.private = 'f'")
-    else
-      joins("LEFT OUTER JOIN watchlists ON watchlists.id = stories.watchlist_id").where("watchlists.id IS NULL OR watchlists.private = 'f'")
-    end
+    query = eager_load(:library_entry)
+            .where("library_entries.id IS NULL OR library_entries.private = 'f'")
```

Comment:
My understanding is that `or` is not possible in the ActiveRecord::Relation DSL, that you have to drop down to ARel or SQL for it.  Also it would be `private: false` since `'f'` is just the way they do false in Postgres